tools/lkl/lib/hijack/hijack.c

```diff
@@ -283,3 +283,20 @@ int epoll_ctl(int epollfd, int op, int fd, struct epoll_event *event)
 
 	return lkl_call(__lkl__NR_epoll_ctl, 4, epollfd, op, fd, event);
 }
+
+int eventfd(unsigned int count, int flags)
+{
+	return lkl_sys_eventfd2(count, flags);
```

Comment:
So MT safe != reentrant? I figured `dlsym` was thread safe and then setting a pointer to the result of `dlsym` would be one `movl` (i.e. a word-sized write), I'm surprised that there are architectures that don't guarantee that to be atomic.

Depending on what we believe to be atomic I think we can get away with less locking, anyway, something like:


    if(!host_foo) {
       acquire(host_foo_lock);
           if(!host_foo) {
             local_foo = dlsym('foo');
             host_foo = local_foo;
          }
      release(host_foo_lock);
    }